### ISSUE TITLE
Excessive input lag with older GFE versions

### ISSUE THREAD

**Issue by oreyg** [moonlight-stream/moonlight-qt#68]
Host:
Windows 10
Intel Core i7 5820k,
gtx 980 ti, GeForce Experience 3.12.0.84,
16 GB RAM

Client:
Windows 10
Intel i5-4200U,
GT 750m (set as default in Nvidia Control Panel),
6 GB RAM

Settings:
stock settings on both (720p 10mb),
100 mb/s link between PC's, 2ms ping, wired

To reproduce:
Grab stock moonlight-qt and moonlight-chrome
Add explorer.exe to the game list on Host machine
Open explorer.exe from moonlight-qt and moonlight-chrome (Or any game where you can see the cursor, if you have any installed)
Notice the difference (or the lack of it, in my case its huge)

Observed behavior:
moonlight-qt 0.4.0 ~250 ms delay
moonlight-chrome 0.9.1 <20 ms delay

Expected behavior:
moonlight-qt performing same or better than moonlight-chrome

moonlight-qt logs:
[moonlight-qt.log](https://github.com/moonlight-stream/moonlight-qt/files/2363877/moonlight-qt.log)

**Comment by cgutman** [comment]
Can you add the information from the Bug Report issue template (which also has details on how to get logs)? https://github.com/moonlight-stream/moonlight-qt/issues/new?template=bug_report.md

**Comment by cgutman** [comment]
Can you upload your Moonlight Qt log? Don't worry about Moonlight Chrome, its logs are useless for performance analysis.

**Comment by oreyg** [comment]
@cgutman upd

**Comment by oreyg** [comment]
Another run
[Moonlight-1536458449.log](https://github.com/moonlight-stream/moonlight-qt/files/2363885/Moonlight-1536458449.log)

**Comment by cgutman** [comment]
@oreyg Thanks. It looks like for some reason your GPU driver (GT 750M) claims to support HEVC even though full HEVC decode was only added in the 1000-series. I'll need to figure out why that's happening.

Please try opening NVIDIA Control Panel and force Moonlight to run on your integrated GPU. If that doesn't fix it, set the codec setting in Moonlight to "Force H.264".

**Comment by oreyg** [comment]
Another run: 60 fps, all auto (it's cleaner than the first log, because I was tweaking settings back then)
[moonlight-qt.log](https://github.com/moonlight-stream/moonlight-qt/files/2363888/moonlight-qt.log)

**Comment by oreyg** [comment]
@cgutman thats probably me tweaking things, sorry
Switching to 'Force H.264' doesn't seem to resolve this problem

**Comment by cgutman** [comment]
Can you upload a log with Force H.264 set?

**Comment by oreyg** [comment]
[Moonlight-1536458963.log](https://github.com/moonlight-stream/moonlight-qt/files/2363897/Moonlight-1536458963.log)
720p 10mbit, Force H.264

**Comment by cgutman** [comment]
Hmm, that log looks fine. We only used 2 decoder surfaces which means the latency should be less than 2 frames.

Can you try forcing Moonlight to run on the iGPU and upload a log of that?

**Comment by oreyg** [comment]
@cgutman
[Moonlight-1536459168.log](https://github.com/moonlight-stream/moonlight-qt/files/2363903/Moonlight-1536459168.log)
Intel HD Graphics, 720p 10mbit, Force H.264

**Comment by cgutman** [comment]
Can you try updating to Moonlight v0.4.0?

If that fails, try updating GeForce Experience to 3.14.1 (latest).

**Comment by oreyg** [comment]
I'm using Moonlight v0.4.0
With latest GeForce Experience its working perfectly, thanks.

**Comment by cgutman** [comment]
@oreyg Your logs looks like it's v0.3.0 (v0.4.0 was just released within the last couple hours)

```
00:00:01 - Qt Debug: Current Moonlight version: "0.3.0"
00:00:02 - Qt Debug: Found update manifest match for current platform
00:00:02 - Qt Debug: Latest version of Moonlight for this platform is: "0.4.0"
00:00:02 - Qt Debug: Update available
```
Good to see that the GeForce Experience update fixed it, though I suspect there must be a bug in the moonlight core library causing poor performance with the older GFE version. I'll keep this ticket open to investigate that.

**Comment by oreyg** [comment]
@cgutman Oops, you're right, I was using version 3.0

**Comment by cgutman** [comment]
It does use the same code but it's being updated regularly with improvements so not all the clients have the exact same version.

**Comment by whatisc** [comment]
I can confirm this is a major issue with the QT version.

There is significant delay in the QT version versus the chrome one.

**Comment by cgutman** [comment]
@whatisc this issue in particular is about older versions of GeForce Experience. Please post your GeForce Experience version and retest with GFE 3.14.1.

If the issue persists, it's not the same bug. Please file a new bug report (including logs) in that case.

**Comment by cgutman** [comment]
The issue turned out to be missing mouse move batching in the Qt client. I just got unlucky that GFE 3.14.1 actually fixed the bug that caused Moonlight to need to batch mouse move events, so I never saw the issue myself.

**Comment by cgutman** [comment]
Fix released in v0.5.0 (Beta 2).
